library/std/src/ffi/os_str.rs

```diff
@@ -1051,6 +1051,37 @@ impl_cmp!(Cow<'a, OsStr>, OsStr);
 impl_cmp!(Cow<'a, OsStr>, &'b OsStr);
 impl_cmp!(Cow<'a, OsStr>, OsString);
 
+macro_rules! impl_partialeq_str {
+    ($lhs:ty, $rhs: ty) => {
+        #[stable(feature = "path_os_str_cmp_str", since = "1.48.0")]
+        impl<'a, 'b> PartialEq<$rhs> for $lhs {
+            #[inline]
+            fn eq(&self, other: &$rhs) -> bool {
+                self.to_str() == Some(other.as_ref())
```

Comment:
`to_str` has the check the entire `OsStr` is valid UTF-8 but you don't need to do that as can be seen in `impl PartialEq<str> for OsStr`: https://github.com/rust-lang/rust/blob/e88e908e66cd1e6e30d789b37bcd774951d01856/library/std/src/ffi/os_str.rs#L950-L956

Reply:
I fixed this up in a new commit (I can squash later).